pkg/cmd/oauth-apiserver/cmd.go

```diff
@@ -111,5 +116,46 @@ func (o *OAuthAPIServerOptions) NewOAuthAPIServer() (*apiserver.OAuthAPIServer,
 		return nil, err
 	}
 
+	etcdOptions := o.RecommendedOptions.Etcd
+	etcdOptions.DefaultStorageMediaType = "application/vnd.kubernetes.protobuf"
+
+	storageFactory := apiserverstorage.NewDefaultStorageFactory(
+		etcdOptions.StorageConfig,
+		etcdOptions.DefaultStorageMediaType,
+		legacyscheme.Codecs,
+		apiserverstorage.NewDefaultResourceEncodingConfig(legacyscheme.Scheme),
+		&serverstorage.ResourceConfig{},
+		specialDefaultResourcePrefixes,
+	)
+
+	if len(etcdOptions.EncryptionProviderConfigFilepath) != 0 {
+		transformerOverrides, err := encryptionconfig.GetTransformerOverrides(etcdOptions.EncryptionProviderConfigFilepath)
+		if err != nil {
+			return nil, err
+		}
+		for groupResource, transformer := range transformerOverrides {
+			storageFactory.SetTransformer(groupResource, transformer)
+		}
+	}
+	if err := o.RecommendedOptions.Etcd.ApplyWithStorageFactoryTo(storageFactory, &serverConfig.GenericConfig.Config); err != nil {
+		return nil, err
+	}
+
 	return serverConfig.Complete().New(genericapiserver.NewEmptyDelegate())
 }
+
+// specialDefaultResourcePrefixes are a custom storage prefixes (we must be backward compatible with OpenShift API)
+var specialDefaultResourcePrefixes = map[schema.GroupResource]string{
+	{Resource: "oauthaccesstokens"}:                                 "oauth/accesstokens",
```

Comment:
the non-grouped variants are probably not necessary anymore.

Reply:
okay, will remove.